tests/integration/pox/stacking/direct-and-pooled-increase-1.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { DevnetNetworkOrchestrator } from "@hirosystems/stacks-devnet-js";
+import { StacksTestnet } from "@stacks/network";
+import { Accounts } from "../../constants";
+import {
+  buildDevnetNetworkOrchestrator,
+  getNetworkIdFromEnv,
+  waitForStacksTransaction,
+} from "../../helpers";
+import { waitForNextRewardPhase } from "../helpers";
+import { broadcastStackSTX } from "../helpers-direct-stacking";
+import {
+  broadcastDelegateSTX,
+  broadcastDelegateStackIncrease,
+} from "../helpers-pooled-stacking";
+
+describe("testing stacker who is also a pool under epoch 2.1", () => {
+  let orchestrator: DevnetNetworkOrchestrator;
+  let timeline = {
+    epoch_2_0: 100,
+    epoch_2_05: 102,
+    epoch_2_1: 106,
+    pox_2_activation: 109,
+  };
+  const fee = 1000;
+
+  beforeAll(() => {
+    orchestrator = buildDevnetNetworkOrchestrator(getNetworkIdFromEnv());
+    orchestrator.start();
+  });
+
+  afterAll(() => {
+    orchestrator.terminate();
+  });
+
+  it("using direct stacking and delegate-stack-increase throws ArithmeticUnderflow,", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Wait for Stacks genesis block
+    await orchestrator.waitForNextStacksBlock();
+    // Wait for block N+1 where N is the height of the next reward phase
+    await waitForNextRewardPhase(network, orchestrator, 1);
+
+    const blockHeight = timeline.pox_2_activation + 1;
+    const cycles = 1;
+
+    // Alice stacks 90m STX
+    let response = await broadcastStackSTX(
+      2,
+      network,
+      90_000_000_000_000,
+      Accounts.WALLET_1,
+      blockHeight,
+      cycles,
+      fee,
+      0
+    );
+    expect(response.error).toBeUndefined();
+
+    // Bob delegates 80m to Alice address
+    response = await broadcastDelegateSTX(
+      2,
+      network,
+      Accounts.WALLET_2,
+      fee,
+      0,
+      80_000_000_000_000,
+      Accounts.WALLET_1
+    );
+    expect(response.error).toBeUndefined();
+    let [block, tx] = await waitForStacksTransaction(
+      orchestrator,
+      response.txid
+    );
+    expect(tx.success).toBeTruthy();
+
+    // Alice increase delegation by 80m
+    response = await broadcastDelegateStackIncrease(
+      2,
+      network,
+      Accounts.WALLET_1,
+      fee,
+      1,
+      Accounts.WALLET_2,
+      Accounts.WALLET_1,
+      80_000_000_000_000
+    );
+    expect(response.error).toBeUndefined();
+    [block, tx] = await waitForStacksTransaction(orchestrator, response.txid);
+    // throws ArithmeticUnderflow
```

Comment:
can we check this throw with `expect()` ?

Reply:
No, the throw is from the Clarity VM and is only logged in docker logs.
@obycode suggested to do a grep on it if really necessary. However, today it is not high priority. Or you have other ideas?

Reply:
ok, maybe it can pop in clarinet testing, we will see.

Reply:
Good point. We can avoid creating the same tests twice.

For pox, integration tests are necessary whenever a tests uses `stx-account` and the `reward-cycle-pox-address-list` map.

Maybe we can generate tests from clarinet unit test.. to see whether simnet is different to the actual node? Not sure whether it makes sense. :thinking: